secure/secure.sh

```diff
@@ -0,0 +1,403 @@
+  set_property_core_site 'hadoop.ssl.server.conf' 'ssl-server.xml'
+  set_property_core_site 'hadoop.ssl.client.conf' 'ssl-client.xml'
+  set_property_core_site 'fs.default.name' "hdfs://$MASTER_FQDN"
+  set_property_core_site 'fs.defaultFS' "hdfs://$MASTER_FQDN"
+}
+
+function set_property_hdfs_site() {
+  set_property_in_xml "$HADOOP_CONF_DIR/hdfs-site.xml" $1 $2
+}
+
+function config_hdfs_site() {
+  set_property_hdfs_site 'dfs.block.access.token.enable' 'true'
+  set_property_hdfs_site 'dfs.encrypt.data.transfer' 'true'
+  # Name Node
+  set_property_hdfs_site 'dfs.namenode.keytab.file' "$HADOOP_CONF_DIR/hdfs.keytab"
+  set_property_hdfs_site 'dfs.namenode.kerberos.principal' "hdfs/_HOST@$REALM"
+  set_property_hdfs_site 'dfs.namenode.kerberos.internal.spnego.principal' "HTTP/_HOST@$REALM"
+  # Secondary Name Node
+  set_property_hdfs_site 'dfs.secondary.namenode.keytab.file' "$HADOOP_CONF_DIR/hdfs.keytab"
+  set_property_hdfs_site 'dfs.secondary.namenode.kerberos.principal' "hdfs/_HOST@$REALM"
+  set_property_hdfs_site 'dfs.secondary.namenode.kerberos.internal.spnego.principal' "HTTP/_HOST@$REALM"
+  # Journal Node
+  set_property_hdfs_site 'dfs.journalnode.keytab.file' "$HADOOP_CONF_DIR/hdfs.keytab"
+  set_property_hdfs_site 'dfs.journalnode.kerberos.principal' "hdfs/_HOST@$REALM"
+  set_property_hdfs_site 'dfs.journalnode.kerberos.internal.spnego.principal' "HTTP/_HOST@$REALM"
+  # Data Node
+  # 'dfs.data.transfer.protection' Enable SASL, only supported after version 2.6
+  set_property_hdfs_site 'dfs.data.transfer.protection' 'privacy'
+  unset HADOOP_SECURE_DN_USER  # Required by SASL
+  set_property_hdfs_site 'dfs.datanode.keytab.file' "$HADOOP_CONF_DIR/hdfs.keytab"
+  set_property_hdfs_site 'dfs.datanode.kerberos.principal' "hdfs/_HOST@$REALM"
+  # WebHDFS
+  set_property_hdfs_site 'dfs.webhdfs.enabled' 'true'
+  set_property_hdfs_site 'dfs.web.authentication.kerberos.principal' "HTTP/_HOST@$REALM"
+  set_property_hdfs_site 'dfs.web.authentication.kerberos.keytab' "$HADOOP_CONF_DIR/http.keytab"
+  # HTTPS
+  set_property_hdfs_site 'dfs.http.policy' 'HTTPS_ONLY'
+
+  set_property_hdfs_site 'dfs.namenode.rpc-address' "$MASTER_FQDN:8020"
+}
+
+function set_property_yarn_site() {
+  set_property_in_xml $HADOOP_CONF_DIR/yarn-site.xml $1 $2
+}
+
+function config_yarn_site_and_permission() {
+  # Resource Manager
+  set_property_yarn_site 'yarn.resourcemanager.keytab' "$HADOOP_CONF_DIR/yarn.keytab"
+  set_property_yarn_site 'yarn.resourcemanager.principal' "yarn/_HOST@$REALM"
+  set_property_yarn_site 'yarn.resourcemanager.hostname' "$MASTER_FQDN"
+  # Node Manager
+  set_property_yarn_site 'yarn.nodemanager.keytab' "$HADOOP_CONF_DIR/yarn.keytab"
+  set_property_yarn_site 'yarn.nodemanager.principal' "yarn/_HOST@$REALM"
+  set_property_yarn_site 'yarn.nodemanager.container-executor.class' 'org.apache.hadoop.yarn.server.nodemanager.LinuxContainerExecutor'
```

Comment:
Why use LinuxContainerExecutor (these two lines and container-executor.cfg)

Reply:
According to https://hadoop.apache.org/docs/r2.9.0/hadoop-yarn/hadoop-yarn-site/SecureContainer.html
"On Linux environment the secure container executor is the LinuxContainerExecutor. It uses an external program called the container-executor to launch the container. This program has the setuid access right flag set which allows it to launch the container with the permissions of the YARN application user."